external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -412,45 +429,68 @@ impl CursorContext {
+    /// auto-completed target kind should be, and weather it is part of the use statement.
+    pub fn find_access_chain(&self) -> Option<ChainInfo> {
         use ChainCompletionKind as CT;
         use CursorPosition as CP;
-        let chain_info = match &self.position {
+        match &self.position {
             CP::Exp(sp!(_, exp)) => match exp {
-                P::Exp_::Name(chain) => Some((chain.clone(), CT::All)),
-                P::Exp_::Call(chain, _) => Some((chain.clone(), CT::Function)),
-                P::Exp_::Pack(chain, _) => Some((chain.clone(), CT::Type)),
-                _ => None,
+                P::Exp_::Name(chain) if chain.loc.contains(&self.loc) => {
+                    return Some(ChainInfo::new(chain.clone(), CT::All, false))
+                }
+                P::Exp_::Call(chain, _) if chain.loc.contains(&self.loc) => {
+                    return Some(ChainInfo::new(chain.clone(), CT::Function, false))
+                }
+                P::Exp_::Pack(chain, _) if chain.loc.contains(&self.loc) => {
+                    return Some(ChainInfo::new(chain.clone(), CT::Type, false))
+                }
+                _ => (),
             },
-            CP::Binding(sp!(_, bind)) => {
-                if let P::Bind_::Unpack(chain, _) = bind {
-                    Some((*(chain.clone()), CT::Type))
-                } else {
-                    None
+            CP::Binding(sp!(_, bind)) => match bind {
+                P::Bind_::Unpack(chain, _) if chain.loc.contains(&self.loc) => {
+                    return Some(ChainInfo::new(*(chain.clone()), CT::Type, false))
                 }
-            }
-            CP::Type(sp!(_, ty)) => {
-                if let P::Type_::Apply(chain) = ty {
-                    Some((*(chain.clone()), CT::Type))
-                } else {
-                    None
+                _ => (),
+            },
+            CP::Type(sp!(_, ty)) => match ty {
+                P::Type_::Apply(chain) if chain.loc.contains(&self.loc) => {
+                    return Some(ChainInfo::new(*(chain.clone()), CT::Type, false))
                 }
-            }
-            CP::Attribute(attr_val) => {
-                if let P::AttributeValue_::ModuleAccess(chain) = &attr_val.value {
-                    Some((chain.clone(), CT::All))
-                } else {
-                    None
+                _ => (),
+            },
+            CP::Attribute(attr_val) => match &attr_val.value {
+                P::AttributeValue_::ModuleAccess(chain) if chain.loc.contains(&self.loc) => {
+                    return Some(ChainInfo::new(chain.clone(), CT::All, false))
+                }
+                _ => (),
+            },
+            CP::Use(
+                sp!(
+                    _,
+                    P::Use::Fun {
+                        visibility: _,
+                        function,
+                        ty,
+                        method: _,
+                    }
```

Comment:
```suggestion
                    P::Use::Fun {
                        function,
                        ty,
                        ..
                    }
```

Reply:
Indeed! In my head (for some reason) field names had to be specified in order which they obviously don't...